drivers/modem/modem_cellular.c

```diff
@@ -0,0 +1,1226 @@
+/*
+ * Copyright (c) 2023 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/modem/chat.h>
+#include <zephyr/modem/cmux.h>
+#include <zephyr/modem/pipe.h>
+#include <zephyr/modem/ppp.h>
+#include <zephyr/modem/backend/uart.h>
+#include <zephyr/net/ppp.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/atomic.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(modem_cellular, CONFIG_MODEM_LOG_LEVEL);
+
+#include <string.h>
+#include <stdlib.h>
+
+enum modem_cellular_state {
+	MODEM_CELLULAR_STATE_IDLE = 0,
+	MODEM_CELLULAR_STATE_POWER_ON,
+	MODEM_CELLULAR_STATE_RUN_INIT_SCRIPT,
+	MODEM_CELLULAR_STATE_CONNECT_CMUX,
+	MODEM_CELLULAR_STATE_OPEN_DLCI1,
+	MODEM_CELLULAR_STATE_OPEN_DLCI2,
+	MODEM_CELLULAR_STATE_RUN_DIAL_SCRIPT,
+	MODEM_CELLULAR_STATE_REGISTER,
+	MODEM_CELLULAR_STATE_ROAMING,
```

Comment:
Change this to something like `ON_NETWORK`. Modems don't always roam.

Reply:
Updated to CARRIER_ON, which reflects the state the network interface in this state